src/bin/genemichaels.rs

```diff
@@ -82,14 +89,23 @@ impl FromStr for On {
     }
 }
 
-#[derive(Parser)]
+#[derive(Parser, Clone)]
 struct Args {
     #[arg(help = "Files to format in place; if none specified formats stdin and writes result to stdout")]
     files: Vec<PathBuf>,
-    #[arg(short, long, default_value_t = FormatConfig::default().max_width)]
-    line_length: usize,
     #[arg(short, long, help = "Won't emit any output")]
     quiet: bool,
+    #[arg(
+        short,
+        long,
+        help =
+            "Formats the entire workspace\n  - if your project is single folder, put an empty [workspace] in your Cargo.toml",
+    )]
+    workspace: bool,
+    #[arg(short, long, help = "Formats the current folder")]
+    folder: bool,
```

Comment:
Mind putting this in a separate mr for now? I'm not sure how many people use rust without cargo, but we can see if this will be useful later.  People should be able to do `genemichaels mydir/*` in any case.

Reply:
Ah I see, this is for if you're not in the directory you want to format. Okay, this seems reasonable to me -- super nitpicky, but would you mind renaming it `dir`? I think that's more common for CLI utils.

I don't think the `[workspace]` should be needed, unless there's a reason that people should be including that in their configs.

Reply:
Not quite - this is for formatting just the folder (and its children) you are currently in.

Its useful for flows where its a big workspace and takes appreciable time to format the entire workspace. Happy to discuss naming further.

Reply:
Ah I see... I think the use case for this is ambiguous.  Using `find` or a glob would work just as well... I thought it was for formatting a rust project not in your current directory.

I think I'd prefer for now to omit this and we can add it later when there's a more immediate use case presented.

Reply:
Fair enough, what I can do of course is split it out into its own draft PR and it can sit there until its needed - fwiw I do think its a valid use case, and perhaps it should take an optional `PathBuf` parameter that can override the directory its formatting.

Reply:
As soon as this PR is ready to go I'll move the folder code to a new PR, just let me know.

Reply:
Okay sounds good.

Actually I had one other thought here, what about making `-p` take a path? For the current directory it would be `genemichaels -p .` which is only one more letter, and it supports the case of formatting project dirs as well.

Reply:
Okay sounds good.